client/valor/coretypes.py

```diff
@@ -61,653 +51,141 @@ def _format_filter(filter_by: Optional[FilterType]) -> Filter:
-            "value": self.value,
-            "score": self.score,
-        }
-
-    @classmethod
-    def from_dict(cls, resp) -> Label:
-        """
-        Construct a label from a dictionary.
-
-        Parameters
-        ----------
-        resp : dict
-            The dictionary containing a label.
-
-        Returns
-        -------
-        valor.Label
-        """
-        return cls(**resp)
-
-    def __eq__(self, other) -> bool:
-        """
-        Defines how `Labels` are compared to one another.
-
-        Parameters
-        ----------
-        other : Label
-            The object to compare with the `Label`.
-
-        Returns
-        ----------
-        boolean
-            A boolean describing whether the two objects are equal.
-        """
-        # type mismatch
-        if type(other) is not type(self):
-            return False
-
-        # k,v mismatch
-        if self.key != other.key or self.value != other.value:
-            return False
-
-        # score is None
-        if self.score is None or other.score is None:
-            return (other.score is None) == (self.score is None)
-
-        # scores not equal
-        if is_float(self.score) and is_float(other.score):
-            return bool(np.isclose(self.score, other.score))
-
-        return False
-
-    def __hash__(self) -> int:
-        """
-        Defines how a `Label` is hashed.
-
-        Returns
-        ----------
-        int
-            The hashed 'Label`.
-        """
-        return hash(f"key:{self.key},value:{self.value},score:{self.score}")
-
-    def tuple(self) -> Tuple[str, str, Union[float, np.floating, None]]:
-        """
-        Defines how the `Label` is turned into a tuple.
-
-        Returns
-        ----------
-        tuple
-            A tuple of the `Label's` arguments.
-        """
-        return (self.key, self.value, self.score)
-
-
-class Annotation:
+class GroundTruth(StaticCollection):
     """
-    A class used to annotate `GroundTruths` and `Predictions`.
-
-    Parameters
-    ----------
-    task_type: TaskType
-        The task type associated with the `Annotation`.
-    labels: List[Label], optional
-        A list of labels to use for the `Annotation`.
-    metadata: Dict[str, Union[int, float, str, bool, datetime.datetime, datetime.date, datetime.time]]
-        A dictionary of metadata that describes the `Annotation`.
-    bounding_box: BoundingBox, optional
-        A bounding box to assign to the `Annotation`.
-    polygon: Polygon, optional
-        A polygon to assign to the `Annotation`.
-    raster: Raster, optional
-        A raster to assign to the `Annotation`.
-    embedding: List[float], optional
-        An embedding, described by a list of values with type float and a maximum length of 16,000.
+    An object describing a ground truth (e.g., a human-drawn bounding box on an image).
 
     Attributes
     ----------
-    geometric_area : float
-        The area of the annotation.
+    datum
```

Comment:
nit: add types and descriptions for these attributes?

Reply:
I was following the numpy format for `properties`

> Attributes that are properties and have their own docstrings can be simply listed by name:

https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring

Reply:
Ive changed the static_collection objects to have explicit attributes as they are not properties